### PR TITLE
feat: participant avatar

### PR DESCRIPTION
**Pull request by AndyLnd**

### CODE REVIEW

File: app/script/components/participantAvatar.js [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+ *
+ */
+
+'use strict';
+
+window.z = window.z || {};
+window.z.components = z.components || {};
+
+z.components.ParticipantAvatar = class ParticipantAvatar {
+  constructor(params, component_info) {
```
Comment (gregor) [inline app/script/components/participantAvatar.js]: camelCase

File: app/script/components/participantAvatar.js [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    this.participant = ko.unwrap(params.participant);
+    this.isUser = this.participant instanceof z.entity.User && !this.participant.isBot;
+    this.isService = this.participant instanceof z.integration.ServiceEntity || this.participant.isBot;
+
+    this.delay = params.delay;
+    this.size = params.size;
+    this.element = $(component_info.element);
+
+    if (this.size) {
+      this.element.addClass(`${this.isService ? 'service' : 'user'}-avatar-${this.size}`);
```
Comment (gregor) [inline app/script/components/participantAvatar.js]: how many things depend on this CSS class? Can we move from individual service and user classes to particpant classes for the sizing or do other things depend on it?

File: app/script/components/participantAvatar.js [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+          case this.participant.is_team_member():
+            return '';
+          case this.participant.is_blocked():
+            return 'blocked';
+          case this.participant.is_request():
+            return 'pending';
+          case this.participant.is_ignored():
+            return 'ignored';
+          case this.participant.is_canceled() || this.participant.is_unknown():
+            return 'unknown';
```
Comment (gregor) [inline app/script/components/participantAvatar.js:72]: If we are at it we can turn these into an enum

File: app/style/components/service-avatar.less [hunk 4]
```diff
@@ -17,7 +17,8 @@
  *
  */
 
-service-avatar {
+service-avatar,
+participant-avatar {
```
Comment (gregor) [inline app/style/components/service-avatar.less:21]: Let's cleanup the CSS classes and remove the outdated service and user avatars in the next step

File: app/script/components/participantAvatar.js (hunk 2 above)
Comment (AndyLnd) [reply app/script/components/participantAvatar.js]: the less files will definitely be merged later on.

File: app/style/components/service-avatar.less (hunk 4 above)
Comment (AndyLnd) [reply app/style/components/service-avatar.less:21]: Most definitely!

File: app/script/components/participantAvatar.js (hunk 3 above)
Comment (AndyLnd) [reply app/script/components/participantAvatar.js:72]: They are just used here. So there is no typo hazard.

**Pull request merged**